frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -201,8 +229,27 @@ public long getStartTime() {
     public void shutdown() {
         running.set(false);
         backendChannel.close();
+        ModelManager manager = ModelManager.getInstance();
         if (currentThread != null) {
             currentThread.interrupt();
+            try {
+                if (parentThreads.size() < model.getMinWorkers()) {
+                    // minWorkers is the numWorkers configured for this model. If parent thread group
+                    // has less threads than expected minWorkers, we should restart the workers.
+                    manager.updateModel(
+                            model.getModelName(), model.getMinWorkers(), model.getMinWorkers());
+                }
+                aggregator.sendError(null, "Internal Failure");
+            } catch (WorkerInitializationException wie) {
+                logger.error(
+                        "Error restarting the backend worker for model name "
+                                + model.getModelName());
+            }
+
+            if (manager.getModels().containsKey(model.getModelName())) {
```

Comment:
getModelName() and getModels() called multiple times, better define variables for them